ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryUrlSigningService.java

```diff
@@ -125,7 +125,7 @@ public String getSignedUrl(RestRequest restRequest) throws RestServiceException
             maxUploadSize = RestUtils.getLongHeader(args, RestUtils.Headers.MAX_UPLOAD_SIZE, true);
             break;
           case RestUtils.Headers.CHUNK_UPLOAD:
-            chunkUpload = RestUtils.getBooleanHeader(args, RestUtils.Headers.CHUNK_UPLOAD, true);
+            chunkUpload = RestUtils.getBooleanHeader(args, RestUtils.Headers.CHUNK_UPLOAD, true, false);
```

Comment:
ifu make a new method getOptionalBooleanHeader, we can avoid this cascade

Reply:
most of the cascade makes it hard to read and unrelated to namedblobs

Reply:
I see this as more about being-explicit, not about code cascade. This method is simple enough (just added one more parameter to be explicit about defaultVal), and not that hard to read comparing to others.

Reply:
I updated this in latest commit. please take another look.